### DISCUSSION TITLE
20.9版本的电源管理是怎么回事，平衡模式的代码写的什么鬼

### DISCUSSION THREAD

**Discussion by debuggerx01** [linuxdeepin/developer-center#4264] [Developer Discussion | 开发者讨论]
具体内容看论坛发的帖子吧，懒得再编辑一遍了：
[https://bbs.deepin.org/zh/post/255788](https://bbs.deepin.org/zh/post/255788)

**Comment by BLumia** [comment]
感谢反馈！我看 @xzl01 已经在原帖下答复过了，实际情况可以参考对应的回复。

> 关于你的回复，20版本代码没有全部放到Github这个事情，我之前是没有听说过的，不知道官方有没有过明确的说明，既然现在知道了那也就无所谓了。

这个我简单答复下。实际是主干代码从 20 变为 23 后，20 与 23 就是分叉维护的了。分叉后的 v20 由 20 的团队在公司内部的 Gerrit 维护（发布后的代码仍然可以 apt source 获取，只是没在 GitHub 了），23 则是在 GitHub 上维护（使用主干 master 分支）。

**Reply by debuggerx01** [reply]
ok，了解了

**Reply by debuggerx01** [reply]
那么分叉后的20的代码会全部进入v23吗？

**Reply by Decodetalkers** [reply]
rebase的，现在是都rebase进来了

**Reply by BLumia** [reply]
不会。大部分项目的 v20 与 v23 版本变化很大，基本也是合不了的。极个别项目做了 v20 与 23 版本的同时支持（比如文管），这种会直接用同一份代码。

**Reply by BLumia** [reply]
> rebase的，现在是都rebase进来了

@debuggerx01 问的是 20 与 23 分开维护后的情况。差别蛮大的。没有 rebase 的关系。
